xc/xenops_server_xen.ml

```diff
@@ -2793,6 +2794,20 @@ module Actions = struct
 		(* Finally, discard any device caching for the domid destroyed *)
 		DeviceCache.discard device_cache domid
 
+	let qemu_disappeared di xc xs =
+		match !Xenopsd.action_after_qemu_crash with
+		|None -> ()
+		|Some action ->
+			debug "action-after-qemu-crash=%s" action;
+			if action="poweroff" then (
```

Comment:
It is more efficient to use a match fot string comparisons. And a bit more readable as well:

```ocaml
match action with 
| "poweroff" ->
	(* we do not expect a HVM guest to survive qemu disappearing, so kill the VM *)
	Domain.set_action_request ~xs di.Xenctrl.domid (Some "poweroff") 
| "pause" -> 
	(* useful for debugging qemu *)
	Domain.pause ~xc di.Xenctrl.domid
| _ -> ()
```